### PR TITLE
Add tests directory to static analysis

### PR DESCRIPTION
**Pull request by franmomu**
It also raises from 1 to 2 phpstan level.

### CODE REVIEW

File: phpstan.neon.dist [hunk 1]
```diff
@@ -1,11 +1,12 @@
+includes:
+    - phpstan-baseline.neon
+
 parameters:
-    level: 1
+    level: 2
+
+    bootstrapFiles:
+            - vendor/bin/.phpunit/phpunit/vendor/autoload.php
```
Comment (VincentLanglet) [inline phpstan.neon.dist:8]:
```suggestion
        - vendor/bin/.phpunit/phpunit/vendor/autoload.php
```

File: tests/App/DataFixtures/MongoDB/BookFixtures.php [hunk 2]
```diff
@@ -29,9 +29,6 @@ public function load(ObjectManager $manager): void
         $manager->flush();
     }
 
-    /**
-     * @phpstan-return class-string[]
```
Comment (VincentLanglet) [inline tests/App/DataFixtures/MongoDB/BookFixtures.php:33]: why ?

Comment (franmomu) [reply tests/App/DataFixtures/MongoDB/BookFixtures.php:33]: there is a more specific return type in the interface: https://github.com/doctrine/data-fixtures/blob/4165e20b8f89827aa5a71229541382a276ced8f9/lib/Doctrine/Common/DataFixtures/DependentFixtureInterface.php#L16

**Review by VincentLanglet — APPROVED** [review]

File: tests/AbstractModelManagerTestCase.php [hunk 3]
```diff
@@ -37,13 +38,13 @@ abstract class AbstractModelManagerTestCase extends TestCase
...
     /**
-     * @var ClassMetadataFactory&Stub
+     * @var ClassMetadataFactory&MockObject
      */
     protected $metadataFactory;
 
     protected function setUp(): void
     {
-        $this->metadataFactory = $this->createStub(ClassMetadataFactory::class);
+        $this->metadataFactory = $this->createMock(ClassMetadataFactory::class);
```
Comment (phansys) [inline tests/AbstractModelManagerTestCase.php:47]: Is `expects()` called on this mock?

Comment (VincentLanglet) [reply tests/AbstractModelManagerTestCase.php:47]: I assume it is, in TestCase which extends from this one.

Comment (phansys) [reply tests/AbstractModelManagerTestCase.php:47]: I don't know that file, but I don't want to block this PR.

**Review by phansys — APPROVED** [review]

File: tests/AbstractModelManagerTestCase.php (hunk 3 above)
Comment (VincentLanglet) [reply tests/AbstractModelManagerTestCase.php:47]: It was introduced in commit https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/commit/3d052206d29b1f699f41666495cfdf223d51b279

And there is some expectations: https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/commit/3d052206d29b1f699f41666495cfdf223d51b279#diff-1891e728773a2fd296e39c4326e06a331b13b4db0a2df95affc3405f13bc1a7bR180

**Comment by VincentLanglet** [comment]
Thanks @franmomu

File: tests/AbstractModelManagerTestCase.php (hunk 3 above)
Comment (phansys) [reply tests/AbstractModelManagerTestCase.php:47]: Thank you for pointing.

Comment (franmomu) [reply tests/AbstractModelManagerTestCase.php:47]: By the way, looking at [`InvocationStubber`](https://github.com/sebastianbergmann/phpunit/blob/9553a43f61a85b377085da47be07f5f0f69a2555/src/Framework/MockObject/Builder/InvocationStubber.php) which is what [`Stub::method()` returns](https://github.com/sebastianbergmann/phpunit/blob/9553a43f61a85b377085da47be07f5f0f69a2555/src/Framework/MockObject/Stub.php#L15), if we call any `with*` method, it should be a mock.

**Pull request merged**